src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/AssessmentApiService.kt

```diff
@@ -38,10 +38,10 @@ class AssessmentApiService(
       { it.answers.firstOrNull()?.refAnswerCode }
     ).filterKeys { it != null }
 
-  fun getAssessmentNeeds(assessmentId: String): Map<Need, NeedSeverity?> =
+  fun getAssessmentNeeds(assessmentId: String): Map<Need, NeedSeverity> =
     assessmentApiClient.getAssessmentNeeds(assessmentId)
-      .filter { it.need != null }
-      .associateBy({ it.need!! }, { it.severity })
+      .filter { it.need != null && it.severity != null }
```

Comment:
We default the severity to 0 if it isn't there, but we might as well exclude it here